scripts/ci/guideline_check.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2021 Intel Corporation
+
+import os
+import sh
+import argparse
+import re
+from unidiff import PatchSet
+
+if "ZEPHYR_BASE" not in os.environ:
+    exit("$ZEPHYR_BASE environment variable undefined.")
+
+repository_path = os.environ['ZEPHYR_BASE']
+
+sh_special_args = {
+    '_tty_out': False,
+    '_cwd': repository_path
+}
+
+coccinelle_scripts = ["/scripts/coccinelle/reserved_names.cocci",
+                      "/scripts/coccinelle/same_identifier.cocci",
+                      "/scripts/coccinelle/identifier_length.cocci",
+                      "/scripts/coccinelle/boolean.cocci",
+                      ]
+
+
+def parse_coccinelle(contents: str, violations: dict):
+    reg = re.compile("([a-zA-Z0-9/]*\\.[ch]:[0-9]*)(:[0-9\\-]*: )(.*)")
+    for line in contents.split("\n"):
+        r = reg.match(line)
+        if r:
+            f = r.group(1)
+            if f in violations:
+                violations[f].append(r.group(3))
+            else:
+                violations[r.group(1)] = [r.group(3)]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Check if change requires full twister")
+    parser.add_argument('-c', '--commits', default=None,
+                        help="Commit range in the form: a..b")
+    parser.add_argument("-o", "--output", required=False,
+                        help="Print violation into a file")
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    if not args.commits:
+        exit("missing commit range")
+
+    # pylint does not like the 'sh' library
+    # pylint: disable=too-many-function-args,unexpected-keyword-arg
```

Comment:
Is this command for pylint disable left over from debug perhaps? Or is this a suggestion in special scenarios?

Reply:
nope, this is due a pylint issue.

Reply:
I see the commit adding this says "suppress pylint flase positive on 'sh' library". Does this commented line execute?